src/coreclr/pal/src/file/file.cpp

```diff
@@ -325,34 +325,10 @@ CorUnix::InternalCanonicalizeRealPath(LPCSTR lpUnixPath, PathCharString& lpBuffe
             goto LExit;
         }
 
-        if (! RealPathHelper(pszCwdBuffer, lpBuffer))
+        if (!RealPathHelper(pszCwdBuffer, lpBuffer))
```

Comment:
Was this reported by Valgrind or a tool like that? The thing is that the lpBuffer is initialized to an empty string (the default constructor of the PathCharString initializes the buffer to zero length zero terminated string.

On macOS, realpath actually sets the lpBuffer even in case of an error, the doc says:
> The realpath() function returns resolved_name on success.  
    If an error occurs, realpath() returns NULL and resolved_name
    contains the pathname which caused the problem.

It seems that the removed handling should stay, but be defined only for HOST_OSX )

Reply:
Yes, `valgrind` flagged this.

The default constructor doesn't seem to `null`-terminate:

https://github.com/dotnet/runtime/blob/78be35936a48414df83460dadf9003e053499f84/src/coreclr/pal/src/include/pal/stackstring.hpp#L110-L113

And `RealPathHelper` doesn't seem to do anything to `lpBuffer` when it returns `FALSE`:

https://github.com/dotnet/runtime/blob/78be35936a48414df83460dadf9003e053499f84/src/coreclr/pal/src/file/file.cpp#L263-L275

Reply:
Ah, I am sorry, I've mistakenly looked few lines below and saw the NullTerminate there. And I've missed the fact that the helper doesn't set the lpBuffer in case of null being returned. So your change is correct.